lib/database/game.php

```diff
@@ -518,22 +518,28 @@ function getAchievementIDs($gameID)
     return $retVal;
 }
 
-function getGameIDFromTitle($gameTitleIn, $consoleID)
+function getGameFromTitle($gameTitleIn, $consoleID)
 {
-    $gameTitle = str_replace("'", "''", $gameTitleIn);
+    $gameTitle = sanitizeTitle($gameTitleIn);
     settype($consoleID, 'integer');
 
-    $query = "SELECT ID FROM GameData
-              WHERE Title='$gameTitle' AND ConsoleID='$consoleID'";
+    $query = "SELECT gd.ID, gd.Title, gd.ConsoleID, gd.ForumTopicID, IFNULL( gd.Flags, 0 ) AS Flags, gd.ImageIcon, gd.ImageTitle, gd.ImageIngame, gd.ImageBoxArt, gd.Publisher, gd.Developer, gd.Genre, gd.Released, gd.IsFinal, c.Name AS ConsoleName, c.ID AS ConsoleID, gd.RichPresencePatch
```

Comment:
I didn't get why all this data being fetched. If I'm not mistaken all the `getGameFromTitle()` clients are using only the game's `ID` and `Title`.

Reply:
In fact, it is exactly the same code than `getGameFrom` but with another filter.
The performance difference is supposed to be very low (if you have the index on the FK).
 I supposed we can use the function somewhere else with some other attributes needs.

But if you prefer, there are many options instead of duplication: 
* having a common function which takes an extra filter param (where `getGameFromTitle ` and `getGameFrom` will pass title and id filter)
* just add the extra param (but I didn't find a good new name for `getGameIDFromTitle`, any idea?)

Reply:
@meleu Any update? What strategy do you prefer?